__tests__/app.test.js

```diff
@@ -83,4 +80,73 @@ describe("API testing", () => {
         });
     });
   });
+  describe("/api/articles/:article_id", () => {
+    test("return correct article when passed an article id ", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: "2020-07-09T20:11:00.000Z",
+            votes: 100,
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          });
+          expect(Object.keys(body.articles)).toHaveLength(8);
+        });
+    });
+    test("for articles with invalid ids we should expect a 404 error", () => {
+      return request(app)
+        .get("/api/articles/4311462")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Article not found");
+        });
+    });
+  });
+  describe("/api/articles/:article_id/comments", () => {
+    test("returns an array of comments with appropriate properties", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.comments).toEqual({
+            article_id: 1,
+            author: "icellusedkars",
+            body: "I hate streaming noses",
+            comment_id: 5,
+            created_at: "2020-11-03T21:00:00.000Z",
+            votes: 0,
+          });
+        });
+    });
+    test("comments should return the most recent one first ", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.comments).not.toEqual({
+            author: "butter_bridge",
```

Comment:
cool that you make the most of the small test dataset to be really specific about what you're expecting